readthedocs/projects/forms.py

```diff
@@ -706,10 +706,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         # Disable domain manipulation on Update, but allow on Create
-        instance = getattr(self, 'instance', None)
-        if instance and instance.pk:
+        if self.instance.pk:
             self.fields['domain'].disabled = True
 
+        # Remove the https option at creation,
+        # but show it if the domain is already marked as http only,
+        # so users can upgrade.
+        if not self.instance.pk or self.instance.https:
+            self.fields.pop("https")
```

Comment:
Never allowing downgrades seems fine, not sure if it's necessary.. but I can see the security issue, so I'm good with it :+1: Always possible to ask support in a very very very rare case.